### ISSUE TITLE
M1 arm update

### ISSUE THREAD

**Issue by MrKlarki** [yousseb/meld#122]
Hello. Will there be an update for arm m1 processors?

**Comment by pi1ot** [comment]
+1

**Comment by yousseb** [comment]
There will be. I'm just currently extremely busy. I'm also kind of waiting on upstream meld to support GTK+ 4.0 as I'm very disappointed with GTK+ 3.0 performance on macOS. The GTK+ 4.0 brings OpenGL support https://blogs.gnome.org/chergert/2020/12/15/gtk-4-got-a-new-macos-backend-now-with-opengl/ which will be a game changer for the meld GUI. Anything before then is not really useful from a performance point of view.

**Comment by DreamDevourer** [comment]
Bad news, according to the Apple Developers video documentations OpenGL support was considered deprecated since 2019 and will be removed shortly after the final release of macOS 12 Monterey.

**Comment by zexpe** [comment]
Just adding my +1 for an ARM M1 release, even if it is with the current, slow GTK+ 3.0, and thanks for compiling & packaging this great Mac OS version for meld.

**Comment by nkoroste** [comment]
+1 would love to see this

**Comment by yousseb** [comment]
> Bad news, according to the Apple Developers video documentations OpenGL support was considered deprecated since 2019 and will be removed shortly after the final release of macOS 12 Monterey.

Don't worry - I'm considering using Google's ANGLE. We just need for Meld to port to GTK 4. I've been doing some experiments - and believe me, they consume time like crazy - one switch and you have to compile for hours. So this seems to be doable. Unfortunately, M1 is not going to be easy for me to try, but I will try to make two builds (x64 & M1) for the next release. It's just pointless at this point to do anything performance related when the GTK quartz renderer is as slow. 

I may, however, create a build that updates Meld to latest stable release - hopefully if I get some time during the coming holidays.

**Comment by zexpe** [comment]
Thanks for the update, and offering to try to make a M1 build of the latest stable release - I'm happy to test it - much appreciated!

**Comment by elias-putz-kiwigrid** [comment]
+1
Would also volunteer for testing any beta build with my MacBook Pro (16-inch, 2021).

**Comment by fangkecn** [comment]
+1

**Comment by IllusionVK** [comment]
Yeah thanks so much @yousseb, your time is very much appreciated because this is such a useful tool, I don't have the skills to help with coding but do let the community know if we can help anywhere.  Thanks again for being awesome 👍

**Comment by ameeno-globality** [comment]
any updates? perhaps build steps?

**Comment by noirchen** [comment]
Any updates on the M1 version?

**Comment by niksy** [comment]
Also looking for this. Maybe we can get build steps for this so we can build it ourselves?

**Comment by jetxr** [comment]
> We just need for Meld to port to GTK 4

Meld's maintainer [states](https://gitlab.gnome.org/GNOME/meld/-/issues/664#note_1426712) "there is no current plan for this and it may well never happen"

**Comment by yousseb** [comment]
Ouch...

OK. I will dedicate some time for this... It will be at least a month.

**Comment by niksy** [comment]
Yeah, this is really unfortunate. Maybe we can proceed with latest Meld on GTK3 for the time being, just so we can have ARM release.

**Comment by chernib** [comment]
Hi! Any updates on the M1 version, please?

**Comment by MakremAyari** [comment]
Hi, Any news about M1 version please?

**Comment by dusanscensny** [comment]
Looking forward to make this working on M2 processors please.

**Comment by zexpe** [comment]
Someone has created an M1 build. You can download the DMG from this pull request: #148

**Comment by yousseb** [comment]
@jetxr Kai posted this a year ago.. But if you look at the bottom of his post, some work has already been done for the gtk-4 support https://gitlab.gnome.org/GNOME/meld/-/issues/664#note_1865650

I'd be OK to take the gtk-4 patch once it get the app running. gtk/pango drawing is super slow on quartz and we really need to use OpenGL (no longer supported by Apple) or Metal (https://developer.apple.com/metal/) for the application to allow us to remove the numerous shortcuts and tweaks that I have to do just to get decent experience while maintaining look & feel.

A lot of discussion had been happening on this thread https://github.com/yousseb/meld/issues/147 and you'll find comments in this regard from people who were able to run stock Meld using Homebrew python/gtk

I'll be trying to make an m1 build soon. Had help from @matthiasdiener who was able to build for m1 locally using the x64 workflow file. I would still need to do some work on the PR for m1 in order to build simultaneously for x64 and m1, but the good news is that we're on path for having a new release for both archs.  In the meantime you could use Matt's build mentioned above (https://github.com/yousseb/meld/pull/148) if the current version stopped working for you with macOS upgrade.

**Comment by jetxr** [comment]
> but the good news is that we're on path for having a new release for both archs.

🤩 Thanks @yousseb, extremely grateful for keeping this going! Can't wait to try out @matthiasdiener's build meanwhile

**Comment by DamianoP** [comment]
In my opinion, this version for ARM does not work very well:
if you try to compare 2 folders with subfolders, it keeps asking the documents folder repeatedly for permission. 
If you provide access to the documents folder, even via the system settings by setting access to the whole disk, Meld still asks for access to the documents folder countless times.
In my opinion, this is not a really usable version, it is good just for comparing individual files

Translated with DeepL.com (free version)

**Comment by DamianoP** [comment]
In my opinion, this version for ARM does not work very well:
if you try to compare 2 folders with subfolders, it keeps asking the documents folder repeatedly for permission. 
If you provide access to the documents folder, even via the system settings by setting access to the whole disk, Meld still asks for access to the documents folder countless times.
In my opinion, this is not a really usable version, it is good just for comparing individual files
